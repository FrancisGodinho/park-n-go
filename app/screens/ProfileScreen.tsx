import { SafeAreaView, StyleSheet, Text } from "react-native";
import React from "react";

type Props = {};

const ProfileScreen = (props: Props) => {
  return (
    <SafeAreaView>
      <Text>ProfileScreen</Text>
    </SafeAreaView>
  );
};

export default ProfileScreen;

const styles = StyleSheet.create({});
