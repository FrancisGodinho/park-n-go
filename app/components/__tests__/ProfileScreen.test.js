import React from "react";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react-native";
import { fireEvent, render } from "@testing-library/react";
import { NavigationContainer } from "@react-navigation/native";
import ProfileScreen from "../../screens/ProfileScreen";

beforeAll((done) => {
  done();
});

afterAll((done) => {
  done();
});

let findTextElement = function (tree, element) {
  console.warn(tree);
  return true;
};

describe("<ProfileScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <NavigationContainer>
          <ProfileScreen />
        </NavigationContainer>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
