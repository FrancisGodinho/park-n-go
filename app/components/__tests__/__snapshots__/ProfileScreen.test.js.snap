// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  accessible={false}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 10,
    }
  }
  testID="screen-view"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#C0C0C0",
            "fontFamily": "Avenir-Black",
            "fontSize": 25,
            "fontWeight": "900",
            "textAlign": "center",
          },
          Object {
            "maxWidth": "90%",
          },
        ]
      }
    >
      undefined
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#404040",
        "borderRadius": 5,
        "flexDirection": "row",
        "marginVertical": 10,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginLeft": 10,
          "width": 30,
        }
      }
    >
      <Text />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      editable={true}
      enablesReturnKeyAutomatically={true}
      keyboardType="default"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="License Plate"
      placeholderTextColor="#A0A0A0"
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Array [
          Object {
            "color": "#C0C0C0",
            "fontFamily": "Avenir-Black",
            "fontSize": 17,
            "fontWeight": "400",
          },
          Object {
            "flex": 1,
            "paddingHorizontal": 10,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#404040",
        "borderRadius": 5,
        "flexDirection": "row",
        "marginVertical": 10,
        "paddingVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginLeft": 10,
          "width": 30,
        }
      }
    >
      <Text />
    </View>
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      editable={true}
      enablesReturnKeyAutomatically={true}
      keyboardType="default"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Credit Card"
      placeholderTextColor="#A0A0A0"
      rejectResponderTermination={true}
      returnKeyType="next"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "color": "#C0C0C0",
            "fontFamily": "Avenir-Black",
            "fontSize": 17,
            "fontWeight": "400",
          },
          Object {
            "flex": 1,
            "paddingHorizontal": 10,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginRight": 10,
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#66A182",
        "borderRadius": 5,
        "marginBottom": 10,
        "marginTop": "auto",
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#C0C0C0",
            "fontFamily": "Avenir-Black",
            "fontSize": 17,
            "fontWeight": "400",
          },
          Object {
            "color": "#E0E0E0",
            "textAlign": "center",
          },
          false,
        ]
      }
    >
      Sign Out
    </Text>
  </View>
</RCTSafeAreaView>
`;
